src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1169,10 +1225,12 @@ core::Error runGrepOperation(const GrepOptions& grepOptions, const ReplaceOption
       cmd << tempFile;
       if (!grepOptions.asRegex())
          cmd << "-F";
-      for (json::Value filePattern : grepOptions.includeFilePatterns())
+      addDirectoriesToCommand(grepOptions.packageFlag(), dirPath, &cmd);
+
+      if (grepOptions.packageFlag() && !grepOptions.includeArgs().empty())
       {
-         cmd << shell_utils::EscapeFilesOnly << "--" << shell_utils::EscapeAll;
-         cmd << filePattern.getString();
+         for (std::string arg : grepOptions.includeArgs())
+            LOG_DEBUG_MESSAGE("Unknown include argument: " + arg);
```

Comment:
Nit: to avoid cluttering the logs for each unexpected argument, could it make sense to log `"Unknown include argument(s): " + boost::join(grepOptions.includeArgs(), ", ")` instead?